### PR TITLE
DP-1152 DP-1163 Projectional editor items split and join

### PR DESCRIPTION
**Pull request by aleksei-berezkin**
NB: there is dependent pull request https://github.com/JetBrains/datapad/pull/35. I haven't tested this one separately from the other, so please merge both or neither.

For all explanations see dependent request.

### CODE REVIEW

File: projectional/src/main/java/jetbrains/jetpad/projectional/generic/ItemsSplitterJoiner.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * limitations under the License.
+ */
+package jetbrains.jetpad.projectional.generic;
+
+import jetbrains.jetpad.base.Pair;
+
+/**
+ * Split and join items in projectional synchronizer as alternative to simple insert-new and delete.
+ */
+public interface ItemsSplitterJoiner<SourceItemT, ViewT> {
```
Comment (timzam) [inline projectional/src/main/java/jetbrains/jetpad/projectional/generic/ItemsSplitterJoiner.java]: How about renaming it to SplitJoinHandler?

Comment (aleksei-berezkin) [inline projectional/src/main/java/jetbrains/jetpad/projectional/generic/ItemsSplitterJoiner.java]: @timzam Good idea

**Comment by aleksei-berezkin** [comment]
Changes in last commits:
* `ItemSplitterJoiner` -> `SplitJoinHandler`
* Merged from master

**Comment by timzam** [comment]
I think you referenced the wrong ticket. It should be DP-953 not DP-1163.

**Comment by timzam** [comment]
Looks good to me.

**Pull request merged**